src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.query;
+
+import com.microsoft.spring.data.gremlin.common.GremlinFactory;
+import com.microsoft.spring.data.gremlin.conversion.GremlinScriptFactory;
+import com.microsoft.spring.data.gremlin.conversion.GremlinSource;
+import com.microsoft.spring.data.gremlin.exception.GremlinInsertionException;
+import com.microsoft.spring.data.gremlin.repository.support.GremlinEntityInformation;
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.apache.tinkerpop.gremlin.driver.Result;
+import org.apache.tinkerpop.gremlin.driver.exception.ResponseException;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.List;
+
+public class GremlinTemplate implements GremlinOperations, ApplicationContextAware {
+
+    private final GremlinFactory gremlinFactory;
+    private final MappingGremlinConverter mappingConverter;
+    private ApplicationContext context;
+
+    public GremlinTemplate(@NonNull GremlinFactory factory, @NonNull MappingGremlinConverter converter) {
+        this.gremlinFactory = factory;
+        this.mappingConverter = converter;
+    }
+
+    @Override
+    public void setApplicationContext(@NonNull ApplicationContext context) throws BeansException {
+        this.context = context;
+    }
+
+    @Override
+    public void deleteAll() {
+        final Client client = this.gremlinFactory.getGremlinClient();
+
+        client.submit(new GremlinScriptEdgeDropLiteral().generateScript(null)).all().join();
+        client.submit(new GremlinScriptVertexDropLiteral().generateScript(null)).all().join();
```

Comment:
What's this generateScript(null)?

Reply:
when we want to delete(drop in gremlin)  the whole graph, we do not need to know the information from domain. Just delete all. So here we do not need any gremlinsource, and passing null instead. It use the same interface as insert.

Reply:
OK, thanks, not sure whether it's caused by I know too little about this gremlin thing, or it's not straight forward enough for the reader.